dev/validate-examples/main.go

```diff
@@ -0,0 +1,117 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"time"
+
+	yaml "gopkg.in/yaml.v2"
+)
+
+type Config struct {
+	ValidationCmd  string `yaml:"validation_cmd"`
+	TeardownCmd    string `yaml:"teardown_cmd"`
+	Timeout        int    `yaml:"timeout"`
+	ExpectedOutput string `yaml:"expected_output"`
+}
+
+func (c *Config) LoadFrom(path string) error {
+	file, err := ioutil.ReadFile(path)
+	if err != nil {
+		return err
+	}
+	err = yaml.Unmarshal(file, c)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func main() {
+	walkErr := filepath.Walk(".", func(exampleConfig string, f os.FileInfo, err error) error {
+		matches, _ := filepath.Match(".validate_example*.yml", f.Name())
+		if matches {
```

Comment:
it's always a good idea to keep a happy path at left ;) 
good article about that: https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88

Reply:
Totally agree, changing